x/feegrant/keeper/keeper.go

```diff
@@ -232,7 +233,7 @@ func (k Keeper) UseGrantedFees(ctx context.Context, granter, grantee sdk.AccAddr
 		return err
 	}
 
-	remove, err := grant.Accept(ctx, fee, msgs)
+	remove, err := grant.Accept(context.WithValue(ctx, corecontext.EnvironmentContextKey, k.Environment), fee, msgs)
```

Comment:
Can we add a disclaimer like this: https://github.com/cosmos/cosmos-sdk/pull/20502/files#diff-efb087e4c9e88b16261eeaec6a655f7260b31fb74709a75774a30309a45de3edR35 in feegrant changelog?